src/Tracy/Logger.php

```diff
@@ -117,87 +132,42 @@ protected function formatLogLine($message, $exceptionFile = null)
 
 
 	/**
-	 * @param  \Exception|\Throwable
-	 * @return string
+	 * @deprecated
 	 */
 	public function getExceptionFile($exception)
 	{
-		while ($exception) {
-			$data[] = [
-				get_class($exception), $exception->getMessage(), $exception->getCode(), $exception->getFile(), $exception->getLine(),
-				array_map(function ($item) { unset($item['args']); return $item; }, $exception->getTrace()),
-			];
-			$exception = $exception->getPrevious();
-		}
-		$hash = substr(md5(serialize($data)), 0, 10);
-		$dir = strtr($this->directory . '/', '\\/', DIRECTORY_SEPARATOR . DIRECTORY_SEPARATOR);
-		foreach (new \DirectoryIterator($this->directory) as $file) {
-			if (strpos($file->getBasename(), $hash)) {
-				return $dir . $file;
-			}
-		}
-		return $dir . 'exception--' . @date('Y-m-d--H-i') . "--$hash.html"; // @ timezone may not be set
+		return $this->blueScreenLogger->getExceptionFile($exception);
 	}
 
 
 	/**
-	 * Logs exception to the file if file doesn't exist.
-	 * @param  \Exception|\Throwable
-	 * @return string logged error filename
+	 * @deprecated
 	 */
 	protected function logException($exception, $file = null)
 	{
-		$file = $file ?: $this->getExceptionFile($exception);
-		$bs = $this->blueScreen ?: new BlueScreen;
-		$bs->renderToFile($exception, $file);
-		return $file;
+		$reflection = new \ReflectionMethod($this->blueScreenLogger, 'logException');
+		$reflection->setAccessible(true);
+		return $reflection->invoke($this->blueScreenLogger, $exception, $file);
 	}
 
 
 	/**
-	 * @param  string|\Exception|\Throwable
-	 * @return void
+	 * @deprecated
 	 */
 	protected function sendEmail($message)
 	{
-		$snooze = is_numeric($this->emailSnooze)
-			? $this->emailSnooze
-			: @strtotime($this->emailSnooze) - time(); // @ timezone may not be set
-
-		if ($this->email && $this->mailer
-			&& @filemtime($this->directory . '/email-sent') + $snooze < time() // @ file may not exist
-			&& @file_put_contents($this->directory . '/email-sent', 'sent') // @ file may not be writable
-		) {
-			call_user_func($this->mailer, $message, implode(', ', (array) $this->email));
-		}
+		$reflection = new \ReflectionMethod($this->mailLogger, 'sendEmail');
```

Comment:
Why is reflection needed? If MailLogger is not interface, it might have sendEmail public. WDYT?